src/main/resources/db/migration/0300/V320__AdHocPlotObservations.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TABLE tracking.observation_types (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL UNIQUE
+);
+
+ALTER TABLE tracking.observations ADD COLUMN is_ad_hoc BOOLEAN DEFAULT FALSE NOT NULL;
+ALTER TABLE tracking.observations ADD COLUMN observation_type_id INTEGER REFERENCES tracking.observation_types;
+ALTER TABLE tracking.observations ADD CONSTRAINT ad_hoc_observation_type CHECK (
+    (is_ad_hoc = FALSE AND observation_type_id IS NULL)
+    OR (is_ad_hoc = TRUE AND observation_type_id IS NOT NULL)
+);
```

Comment:
What would you think about making observation type non-nullable and setting it to "Monitoring" for existing (and new non-ad-hoc) observations?

My sense is that observation type and ad-hocness are independent variables. We've already talked about possibly having assigned biomass observations, for example.